formats/json/jvmMain/src/kotlinx/serialization/json/internal/JsonToWriterStringBuilder.kt

```diff
@@ -0,0 +1,41 @@
+package kotlinx.serialization.json.internal
+
+import java.io.OutputStream
+import java.io.Writer
+import java.nio.charset.Charset
+
+
+internal class JsonToWriterStringBuilder(private val writer: Writer) : JsonStringBuilder(
+    // maybe this can also be taken from the pool, but currently initial char array size there is 128, which is too low.
+    CharArray(BATCH_SIZE)
```

Comment:
It seems to be pretty huge allocation (e.g. it always misses TLAB). Could you please ensure it doesn't dominate small objects serialization?

If so, it's worth either reducing its size or pool a few instances of this